### PR TITLE
Add Agent Framework DevUI into the aiagent-webapi template

### PR DESCRIPTION
**Pull request by jeffhandley**
The Microsoft.Agents.AI.DevUI package will soon be published, and it will be enabled by default in the aiagent-webapi project template, with the option to disable it. When DevUI is enabled, the OpenAI Responses and Conversations need to be added into the app.

This PR also improves the project template execution test output to emit the stdout and stderr when not empty, as well as an expanded test description that does not get truncated (like the `[Theory]` args do). As part of that, an issue is fixed in the project template projects where the `ExecutionTestSandbox` folders were being included in the projects, causing build errors in the projects after execution tests had been run.

_This cannot be merged until DevUI is published and the package versions are updated in `src/ProjectTemplates/GeneratedContent.targets`_
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/extensions/pull/7026)

### CODE REVIEW

File: test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/WebApiAgentTemplateExecutionTests.cs [hunk 1]
```diff
@@ -55,6 +55,7 @@ public async Task CreateRestoreAndBuild(params string[] args)
     private static readonly (string name, string[] values)[] _templateOptions = [
         ("--provider",         ["azureopenai", "githubmodels", "ollama", "openai"]),
         ("--managed-identity", ["true", "false"]),
+        ("--no-devui",         ["true", "false"]),
```
Comment (stephentoub) [inline test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/WebApiAgentTemplateExecutionTests.cs]: I'm just curious about choosing to make this an option, since it's adding complexity throughout the template. We think it's important to be able to generate apps without this?

Comment (jeffhandley) [reply test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/WebApiAgentTemplateExecutionTests.cs]: Fair question! I will chat with others offline to decide if it's valuable to have the option. Thanks.

Comment (jeffhandley) [reply test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/WebApiAgentTemplateExecutionTests.cs]: @danroth27 and I agreed with your feedback. I removed the parameter. Good call; thanks!

File: src/ProjectTemplates/Microsoft.Agents.AI.ProjectTemplates/src/WebApiAgent/WebApiAgent-CSharp/Properties/launchSettings.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
     "http": {
       "commandName": "Project",
       "dotnetRunMessages": true,
-      "launchBrowser": false,
+      "launchBrowser": true,
+      "launchUrl": "devui/",
```
Comment (danroth27) [inline src/ProjectTemplates/Microsoft.Agents.AI.ProjectTemplates/src/WebApiAgent/WebApiAgent-CSharp/Properties/launchSettings.json:8]: Is the trailing slash needed?

File: test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/Snapshots/aiagent-webapi.GitHubModels.verified/aiagent-webapi/Program.cs [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
 //   cd this-project-directory
 //   dotnet user-secrets set GitHubModels:Token YOUR-GITHUB-TOKEN
-var credential = new ApiKeyCredential(builder.Configuration["GitHubModels:Token"] ?? throw new InvalidOperationException("Missing configuration: GitHubModels:Token. See README for details."));
-var openAIOptions = new OpenAIClientOptions { Endpoint = new Uri("https://models.inference.ai.azure.com") };
-
-var chatClient = new OpenAIClient(credential, openAIOptions)
-    .GetChatClient("gpt-4o-mini").AsIChatClient();
+var chatClient = new ChatClient(
+        "gpt-4o-mini",
+        new ApiKeyCredential(builder.Configuration["GitHubModels:Token"] ?? throw new InvalidOperationException("Missing configuration: GitHubModels:Token.")),
```
Comment (danroth27) [inline test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/Snapshots/aiagent-webapi.GitHubModels.verified/aiagent-webapi/Program.cs]: Should we use "GITHUB_TOKEN" for this config value? It seems like it's pretty common to use "GITHUB_TOKEN" for personal access tokens. For example, that's what the [Aspire GitHub Models Integration](https://learn.microsoft.com/dotnet/aspire/github/github-models-integration?tabs=dotnet-cli#default-api-key-parameter) looks for by default. And it's what GitHub Actions use for [authentication in workflows](https://docs.github.com/en/actions/tutorials/authenticate-with-github_token). I get that the current config value is with the AI Chat Web App template currently uses.

File: src/ProjectTemplates/Microsoft.Agents.AI.ProjectTemplates/src/WebApiAgent/WebApiAgent-CSharp/Properties/launchSettings.json (hunk 2 above)
Comment (jeffhandley) [reply src/ProjectTemplates/Microsoft.Agents.AI.ProjectTemplates/src/WebApiAgent/WebApiAgent-CSharp/Properties/launchSettings.json:8]: The trailing slash isn't _needed_, but if you launch `devui` without, Dev UI immediately redirects you to `devui/` with the slash.

File: test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/Snapshots/aiagent-webapi.GitHubModels.verified/aiagent-webapi/Program.cs (hunk 3 above)
Comment (jeffhandley) [reply test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/Snapshots/aiagent-webapi.GitHubModels.verified/aiagent-webapi/Program.cs]: I snapped to the same config property name we used in the `aichatweb` template, but I think you're right that we should be consistent with the other resources you referenced instead. I'll change this.

Comment (jeffhandley) [reply test/ProjectTemplates/Microsoft.Agents.AI.Templates.IntegrationTests/Snapshots/aiagent-webapi.GitHubModels.verified/aiagent-webapi/Program.cs]: Updated to `GITHUB_TOKEN` and `OPENAI_KEY`, while leaving `AzureOpenAI:Key` and `AzureOpenAI:Endpoint` since that keeps those 2 settings together. I only made this change for the new `aiagent-webapi` template, leaving `aichatweb` alone.

/cc @MackinnonBuck @davidortinau @jongalloway

File: test/ProjectTemplates/Microsoft.Extensions.AI.Templates.IntegrationTests/Microsoft.Extensions.AI.Templates.Tests.csproj [hunk 4]
```diff
@@ -21,12 +21,15 @@
...
     <Compile Include="..\Infrastructure\*.cs" Link="Infrastructure\%(Filename)%(Extension)" />
+    <Compile Remove="ExecutionTestSandbox\**" />
 
     <Compile Remove="Snapshots\**\*.*" />
     <None Include="Snapshots\**\*.*" />
 
     <Compile Remove="ExecutionTestSandbox\**\*.*" />
     <None Include="ExecutionTestSandbox\**\*.*" />
+    <Content Remove="ExecutionTestSandbox\**\*.*" />
+    <EmbeddedResource Remove="ExecutionTestSandbox\**\*.*" />
```
[the remark this answers is not in the document]
Comment (jeffhandley) [reply test/ProjectTemplates/Microsoft.Extensions.AI.Templates.IntegrationTests/Microsoft.Extensions.AI.Templates.Tests.csproj]: Bad merge conflict resolution here. I'm going to just undo the changes to this file and keep the PR localized to Microsoft.Agents.AI.ProjectTemplates without touching Microsoft.Extensions.AI.Templates.

Good catch, Copilot.
